### ISSUE TITLE
Readonly date segment keyboard navigation is non-standard on iOS

### ISSUE THREAD

**Issue by jakeleveroni** [adobe/react-spectrum#6906]
### Provide a general summary of the issue here

When building a readonly date input component using the useDateField hook, the date segments are not keyboard navigable on iOS devices. I tested via browserstack on multiple iOS devices including iPhone 14, iPhone 16, & iPad 10th generation. Readonly date segments are keyboard navigable in other browsers and on other devices as expected, by using the tab key and arrow keys. 

### 🤔 Expected Behavior?

Readonly date inputs should have standard keyboard navigation applied to them across all browsers/devices

### 😯 Current Behavior

Readonly date inputs are not keyboard navigable on iOS devices

### 💁 Possible Solution

This is a real head-scratcher i have not found the root cause

### 🔦 Context

We are building a UI component library that adheres to WCAG accessibility guidelines.

### 🖥️ Steps to Reproduce

I was able to repro here: https://codesandbox.io/p/sandbox/react-aria-template-389r79 on desktop chrome/safari/firefox keyboard navigation is fine. only on iOS based devices is it broken

### Version

3.30.0

### What browsers are you seeing the problem on?

Other

### If other, please specify.

All browsers on iOS only

### What operating system are you using?

iOS 16.4

**Comment by ktabors** [comment]
Can you provide more information on reproduction?

There seems to be a save issue with your codesandbox, i.e. it looks like a template with no changes, so I created https://codesandbox.io/p/sandbox/awesome-ully-rw67p7. I'm able to arrow key left, arrow key right, tab, and shift-tab between date segments using my iPhone 14 running iOS 17.5.1. I'm using an old Apple Magic Keyboard.

**Comment by jakeleveroni** [comment]
@ktabors hey sorry about that, i must have linked the sandbox incorrectly. Here is the one I was using https://codesandbox.io/p/sandbox/compassionate-meitner-ntskgw its quite similar to your example except when I call useDateFieldState I pass a default value and the isReadonly boolean like so:
```
  let state = useDateFieldState({
    ...{ ...props, defaultValue: parseDate("2024-02-04"), isReadOnly: true },
    locale,
    createCalendar,
  });
```

This link should be for the sandbox I tried to originally link: https://codesandbox.io/p/sandbox/compassionate-meitner-ntskgw

**Comment by ktabors** [comment]
I updated my example to include the `isReadOnly` and `defaultValue`. Sorry I forgot to include "isReadOnly" in my example. I passed them in via props to create a difference to see if that matters. Both examples work on my iPhone with my keyboard, I'm not able to reproduce the error.

There were a few seconds where tab key events appeared to stop working when I had the year selected on your example, but I was able to get out of that state using arrow keys. And I've been unable to get back to that. I've reloaded the page and tried a few different paths to get in and out of the date field and DateSegements.

I'm assuming this issue is constant where you tab to or select via touch a DateSegment and you're unable to tab or arrow key away from that segment. There are no reproduction steps beyond this, correct?

@majornista Do you have a setup where you can test this issue?

**Comment by jakeleveroni** [comment]
I think something is up with codesandbox. I can't load any previews as of this morning. I was going to try and send a video of what I'm seeing when previews are working again. It might be worth noting I'm testing via Browserstack I don't have access to a physical device, unfortunately.

**Comment by jakeleveroni** [comment]
Codesandbox is working again, i just tested via browserstack and it seems like the focus navigation is working as expected. Not sure what I was seeing originally since its working now but Ill close this issue out. Thanks @ktabors for looking at this as well.
